### ISSUE TITLE
Clicking clear closes Headless UI modal

### ISSUE THREAD

**Issue by stevebauman** [vueform/multiselect#182]
### Version

* Vue version: 3

### Description

Similarly to https://github.com/vueform/multiselect/issues/148, when you click the clear (X) button inside the input element, it closes the headless UI modal.

I was able to resolve this using a `@click` handler instead of `@mousedown`, which triggers the modal to close:

```vue
<Multiselect>
    <template #clear="{ clear }">
        <span :class="classes.clear" @click="clear">
            <span :class="classes.clearIcon"></span>
        </span>
    </template>
</Multiselect>
```

### Demo

Unable to produce demo due to no available CDN for Headless UI.

**Comment by rizkhal** [comment]
Hy @stevebauman, did you find the solution?

**Comment by stevebauman** [comment]
Hi @rizkhal, yes I mentioned the solution in the description of this issue 👍

**Comment by rizkhal** [comment]
Oh yes, i followed your instructions but still not working, then i added `@click.prevent` and it works fine.. Thanks

**Comment by giusepetroso** [comment]
Hi guys, I'm facing the same error.

First i noticed the issue with selected tags clear button
I fixed it using @click in the "clear" slot
Then I got the same issue on the clear button of the select itself.
Can't find a way to make it work

If anyone have suggestions they'll be appreciated

**Comment by ejirocodes** [comment]
Thanks @stevebauman, you're a genius 👍🏽

**Comment by adamberecz** [comment]
Fixed in `2.4.0` - thanks guys. Please reopen if you still encounter the issue (I was not able to reproduce).
